src/main/java/org/jabref/gui/openoffice/OOBibBase.java

```diff
@@ -158,42 +157,32 @@ public boolean isConnectedToDocument() {
         return xCurrentComponent != null;
     }
 
-    public static XTextDocument selectComponent(List<XTextDocument> list)
-            throws UnknownPropertyException, WrappedTargetException, IndexOutOfBoundsException {
-        String[] values = new String[list.size()];
-        int ii = 0;
-        for (XTextDocument doc : list) {
-            values[ii] = String.valueOf(OOUtil.getProperty(doc.getCurrentController().getFrame(), "Title"));
-            ii++;
-        }
-        JList<String> sel = new JList<>(values);
-        sel.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
-        sel.setSelectedIndex(0);
-        int ans = JOptionPane.showConfirmDialog(null, new JScrollPane(sel), Localization.lang("Select document"),
-                JOptionPane.OK_CANCEL_OPTION);
-        if (ans == JOptionPane.OK_OPTION) {
-            return list.get(sel.getSelectedIndex());
-        } else {
-            return null;
-        }
+    public XTextDocument selectComponent(List<XTextDocument> list) {
+        List<DocumentTitleViewModel> viewModel = list.stream().map(DocumentTitleViewModel::new).collect(Collectors.toList());
+        //this whole method is part of a background task when autodecting instances, so we need to show dialog in FX thread
```

Comment:
This is bound to produce threading issues. The whole javafx concurrency is build around the paradigm that only the `onSucussful` (or `onError`) handler's run code on the UI thread. I hope this issue can be resolved with minor refactorings and chaining tasks.

Reply:
Okay, I will try to refactor that further